packages/health-check/src/checks/critical.js

```diff
@@ -53,6 +54,22 @@ async function uploadCheck(done) {
     data.statusCode = response.statusCode;
     data.up = true;
     data.ip = response.ip;
+
+    // Check file health
+    const responseContent = getResponseContent(response);
+    const skylink = responseContent.skylink;
+    try {
+      await skylinkHealthCheck(skylink, 30, authCookie);
+    } catch (error) {
+      // Reset the up status as the previous successful file upload would have
+      // set this to true.
+      data.up = false;
+      data.statusCode = error.response?.statusCode || error.statusCode || error.status;
+      // Default to the error itself if the message or response are null since
+      // the error can be a simple error string.
+      data.errorMessage = error.message || error;
+      data.errorResponseContent = getResponseContent(error.response) || error;
+    }
```

Comment:
if you move the `await skylinkHealthCheck(...)` call just below `const response = await got.post(...)` then you can skip all that try-catch block since that's duplicating the outer try-catch block.